src/libMessage/Messenger.cpp

```diff
@@ -741,10 +744,13 @@ bool ProtobufToShardingStructureAnnouncement(
         govProposalId = proto_member.powsoln().govdata().proposalid();
         govVoteValue = proto_member.powsoln().govdata().votevalue();
       }
+      // Copy 32 bytes of extraData at most. Validation of the PoW solution will fail later if the extraData was longer.
+      zbytes extraData(proto_member.powsoln().extradata().begin(), proto_member.powsoln().extradata().begin() + 32);
```

Comment:
This will copy junk (and possibly segfault?) if there isn't 32 bytes of extradata ..

Reply:
I think you're right, thanks for spotting this!